lib/vcert.rb

```diff
@@ -56,26 +58,31 @@ def renew(*args)
     # @param [String] zone
     # @param [Integer] timeout
     # @return [Certificate]
-    def request_and_retrieve(req, zone, timeout)
+    def request_and_retrieve(req, zone, timeout: TIMEOUT)
       request zone, req
+      cert = retrieve_loop(req, timeout: timeout)
+      return cert
+    end
+
+    def retrieve_loop(req, timeout: TIMEOUT)
       t = Time.new() + timeout
       loop do
         if Time.new() > t
           LOG.info("Waiting certificate #{req.id}")
           break
         end
-        certificate = retrieve(req)
+        certificate = @conn.retrieve(req)
```

Comment:
don\`t need to specify `@conn.`

Reply:
if you don't you will get endless stack error, already tried